File: problems/1030_Matrix_Cells_In_Distance_Order.cpp
#include <bits/stdc++.h>

using namespace std;

vector< vector<int> > allCellsDistOrder(int R, int C, int r0, int c0) {
  set< pair <int, vector<int> > > coordinates;
  pair< int, vector<int> > pairing;
  vector<int> coordinate;
  for(int i=0; i<R; i++) {
    for(int j=0; j<C; j++) {
      coordinate.clear();
      coordinate.push_back(i);
      coordinate.push_back(j);
      pairing = make_pair(abs(i-r0)+abs(j-c0),coordinate);
      coordinates.insert(pairing);
    }
  }
  vector< vector<int> > ans;
  for(auto pair : coordinates) {
    ans.push_back(pair.second);
  }
  return ans;
}

int main() {
  vector< vector<int> > list = allCellsDistOrder(2,3,1,3);
  for(auto item : list) {
    for(auto num : item) {
      cout << num << " ";
      
    }
  }
}

File: problems/1030_Matrix_Cells_In_Distance_Order.py
def allCellsDistOrder(R, C, r0, c0):
    coordinates = []
    for i in range(R):
        for j in range(C):
            pair = (abs(i-r0)+abs(j-c0),[i,j])
            print(pair)
            coordinates.append(pair);
    coordinates = sorted(coordinates, key=lambda coordinate: coordinate[0])
    ans = []
    for coordinate in coordinates:
        ans.append(coordinate[1])
    return ans


print(allCellsDistOrder(2,3,1,2))

